src/source/Ice/IceAgent.c

```diff
@@ -2112,6 +2120,10 @@ STATUS iceAgentNominatingStateSetup(PIceAgent pIceAgent)
 
     pIceAgent->stateEndTime = GETTIME() + pIceAgent->kvsRtcConfiguration.iceCandidateNominationTimeout;
 
+    MUTEX_UNLOCK(pIceAgent->lock);
+    locked = FALSE;
+    checkIceAgentStateMachine(pIceAgent);
```

Comment:
In `checkIceAgentStateMachine`, we use `pIceAgent->pStateMachine`, which could be updated in multiple threads right? So, doesnt it need to happen within the mutex locks? Or we dont see this event happening?

Reply:
The pattern isn't consistent between TURN and ICE state machines, but in all the from* function of the ice agent state machine locks occur where relevant. The check function does not edit the pStateMachine.